holoviews/util/locator.py

```diff
@@ -0,0 +1,171 @@
+"""
+Minimal set of functionality of Matplotlib's MaxNLocator to choose contour
+levels without having to have Matplotlib installed.
+"""
+import math
+import numpy as np
+
+
+class _Edge_integer:
+    """
+    Helper for `.MaxNLocator`, `.MultipleLocator`, etc.
+
+    Take floating-point precision limitations into account when calculating
+    tick locations as integer multiples of a step.
+    """
+    def __init__(self, step, offset):
+        """
+        Parameters
+        ----------
+        step : float > 0
+            Interval between ticks.
+        offset : float
+            Offset subtracted from the data limits prior to calculating tick
+            locations.
+        """
+        if step <= 0:
+            raise ValueError("'step' must be positive")
+        self.step = step
+        self._offset = abs(offset)
+
+    def closeto(self, ms, edge):
+        # Allow more slop when the offset is large compared to the step.
```

Comment:
Is slop correct?

Reply:
It is copy-pasted from mpl, I don't wish to understand or change it!